src/components/Documents/DocumentPreview.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { Dimensions, Image, StyleSheet, View } from 'react-native';
+import { useShowDocument } from '../../hooks/DocumentsHooks';
+import { DocumentInterface } from '../../types/Documents';
+import PdfComponent from '../UI/Pdf';
+
+const DocumentPreview: React.FC<{ document: DocumentInterface }> = ({ document }) => {
+  const { previewUrl } = useShowDocument(document.id);
+  const isPreviewUrlEmpty = previewUrl === '';
+
+  return !isPreviewUrlEmpty && document.extension !== 'pdf' ? (
+    <Image style={styles.imageWrapper} source={{ uri: previewUrl }} />
+  ) : !isPreviewUrlEmpty ? (
+    <View style={styles.pdfWrapper}>{document.url && <PdfComponent uri={document.url} />}</View>
+  ) : null;
+};
```

Comment:
I'm not a big fan of nested ternaries, I think I would also use early returns hrer
```jsx
if (!!isPreviewUrlEmpty) return null;

return document.extension !== 'pdf' 
    ?  <Image style={styles.imageWrapper} source={{ uri: previewUrl }} />
    :  <View style={styles.pdfWrapper}>{document.url && <PdfComponent uri={document.url} />}</View>;
```